lib/ramfs/README.md

```diff
@@ -0,0 +1,115 @@
+# RamFS: Unikraft’s RAM-based Virtual Filesystem
+
+The `RamFS` filesystem is a virtual filesystem that uses memory for storage, as the name implies.
+Its advantage is that it is very fast.
+From a simplified perspective, we can look at a file in `RamFS` as a buffer in memory.
+Since this is how `RamFS` represents data, the disadvantage becomes apparent: we can only write data into `RamFS` until we fill up all the memory.
+Moreover, data is not persistent: once we stop the unikernel, memory is freed and data is lost.
+
+## Key Functions and Data Structures
+
+The key structure used is `ramfs_node` defined as:
+
+```c
+struct ramfs_node {
+ /* Next node in the same directory */
+ struct ramfs_node *rn_next;
+ /* First child node */
+ struct ramfs_node *rn_child;
+ /*
+  * Entry type: regular file - VREG, symbolic link - VLNK,
+  * or directory - VDIR
+  */
+ int rn_type;
+ /* Name of the file/directory (NUL-terminated) */
+ char *rn_name;
+ /* Length of the name not including the terminator */
+ size_t rn_namelen;
+ /* Size of the file */
+ size_t rn_size;
+ /* Buffer to the file data */
+ char *rn_buf;
+ /* Size of the allocated buffer */
+ size_t rn_bufsize;
+ /* Last change time */
+ struct timespec rn_ctime;
+ /* Last access time */
+ struct timespec rn_atime;
+ /* Last modification time */
+ struct timespec rn_mtime;
+ /* Node access mode */
+ int rn_mode;
+ /* Whether the rn_buf was allocated in this ramfs_node */
+ bool rn_owns_buf;
+};
+```
+
+This structure contains the `rn_type` field, which refers to the entry type (which can be a regular file - `VREG`, a symbolic link - `VLNK`, or a directory - `VDIR`), the buffer in which the data will be stored (the `rn_buf` field), and its size (`rn_size`).
```

Comment:
```suggestion
This structure contains:

* The `rn_type` field, which refers to the entry type.
It can be a regular file - `VREG`, a symbolic link - `VLNK`, or a directory - `VDIR`
* The  `rn_buf` field, which is the buffer in which the data will be stored
*  The buffer  size, `rn_size`

```
This sounds a little confusing. The meaning of the sentence gets lost because of the long explanation for the `rn_type` field.
I would do this as a list or something similar.